terratorch/datasets/open_sentinel_map.py

```diff
@@ -0,0 +1,222 @@
+import json
+import random
+import re
+from pathlib import Path
+
+import albumentations as A  # noqa: N812
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn.functional as F  # noqa: N812
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+from torchgeo.datasets import NonGeoDataset
+
+from terratorch.datasets.utils import to_tensor, pad_numpy
+
+
+class OpenSentinelMap(NonGeoDataset):
+    def __init__(
+        self,
+        data_root: str,
+        split: str = "train",
+        bands: list[str] | None = None,
+        transform: A.Compose | None = None,
+        interpolate_and_concat_bands: bool = True,  # noqa: FBT001, FBT002
+        pad_image: int | None = None,
+        truncate_image: int | None = None,
+        target: int = 0,
+        pick_random_pair: bool = True,  # noqa: FBT002, FBT001
+    ) -> None:
+        """
+        Pytorch Dataset class to load samples from the OpenSentinelMap dataset, supporting
+        multiple bands and temporal sampling strategies.
+
+        Args:
+            data_root (str): Path to the root directory of the dataset.
+            split (str): Dataset split to load. Options are 'train', 'val', or 'test'. Defaults to 'train'.
+            bands (list of str, optional): List of band names to load. Defaults to ['gsd_10', 'gsd_20', 'gsd_60'].
+            transform (albumentations.Compose, optional): Albumentations transformations to apply to the data.
+            interpolate_and_concat_bands (bool): If True, the bands are interpolated and concatenated over time. Default is True.
```

Comment:
Yes, we interpolate the bands spatially (since they have different resolutions) so we can concatenate in the channels axis. after that, the images are temporally stacked. I wiil change it from "interpolate_and_concat_bands" to "spatial_interpolate_and_stack_temporally".